src/main/java/org/eclipse/xpanse/terraform/boot/terraform/TerraformVersionHelper.java

```diff
@@ -0,0 +1,291 @@
+    private static final Pattern TERRAFORM_REQUIRED_VERSION_PATTERN =
+            Pattern.compile(TERRAFORM_REQUIRED_VERSION_REGEX);
+    private static final Pattern TERRAFORM_VERSION_OUTPUT_PATTERN =
+            Pattern.compile("^Terraform\\s+v(\\d{1,3}\\.\\d{1,3}\\.\\d{1,3})\\b");
+    private static final String OS_NAME = System.getProperty("os.name").toLowerCase();
+    private static final String OS_ARCH = System.getProperty("os.arch").toLowerCase();
+
+    @Value("${terraform.install.dir:/opt/terraform}")
+    private String terraformInstallDir;
+    @Resource
+    private SystemCmd systemCmd;
+
+    /**
+     * Get terraform executor path which matched the required version.
+     *
+     * @param requiredVersion the required version
+     * @return return the version of terraform which is matched required, otherwise return null.
+     */
+    public String getExecutorPathMatchedRequiredVersion(String requiredVersion) {
+        // Get path of terraform executor matched required version in the installation dir.
+        File installDir = new File(terraformInstallDir);
+        if (!installDir.exists() || !installDir.isDirectory()) {
+            return null;
+        }
+        File[] terraformExecutors = installDir.listFiles();
+        for (File terraformExecutor : terraformExecutors) {
+            String executorVersion =
+                    getVersionFromExecutorPath(terraformExecutor.getAbsolutePath());
+            if (terraformExecutor.isFile() && isVersionSatisfied(executorVersion, requiredVersion)
+                    && terraformExecutor.canExecute()) {
+                if (checkIfExecutorVersionIsValid(terraformExecutor, requiredVersion)) {
+                    log.info("Terraform with path {} matched required version {} already "
+                            + "installed.", terraformExecutor.getAbsolutePath(), requiredVersion);
+                    return terraformExecutor.getAbsolutePath();
+                }
+            }
+        }
+        return null;
+    }
+
+    private String cleanRequiredVersion(String version) {
+        return version.replaceAll("\\s+", "").toLowerCase()
+                .replaceAll("v", "");
+    }
+
+
+    /**
+     * Get the matched version existed in download url.
+     *
+     * @param downloadBaseUrl the base url of download
+     * @param requiredVersion required version
+     * @return the matched version existed in download url.
+     */
+    public String getMatchedVersionExistedInDownloadUrl(String downloadBaseUrl,
+                                                        String requiredVersion) {
+        try {
+            Document doc = Jsoup.connect(downloadBaseUrl).get();
+            Elements elements = doc.select("a[href]");
+            for (Element element : elements) {
+                String href = element.attr("href");
+                Matcher matcher = Pattern.compile("^/terraform/(\\d{1,3}\\.\\d{1,3}\\.\\d{1,3})/")
```

Comment:
would it make sense to move it to a static field so that it is only compiled only once.